packages/swing-store/src/swingStore.js

```diff
@@ -224,14 +231,14 @@ function makeSwingStore(dirPath, forceReset, options) {
     assert.typeof(start, 'string');
     assert.typeof(end, 'string');
 
-    ensureTxn();
-    const cursor = new lmdb.Cursor(txn, dbi);
-    let key = start === '' ? cursor.goToFirst() : cursor.goToRange(start);
-    while (key && (end === '' || key < end)) {
-      yield key;
-      key = cursor.goToNext();
-    }
-    cursor.close();
+    /** @type {import("lmdb").RangeOptions} */
+    const rangeOptions = {};
+    if (start) rangeOptions.start = start;
+    if (end) rangeOptions.end = end;
+
+    const keys = ensureTxn().getKeys(rangeOptions);
```

Comment:
Is `keys` an iterator, or an array? I want to make sure we aren't holding a gigantic list of keys in RAM here.

Reply:
A proper iterator! With lots of logic regarding transactions boundaries.